public/app/features/explore/TraceView/createSpanLink.tsx

```diff
@@ -80,35 +82,66 @@ function legacyCreateSpanLinkFactory(splitOpenFn: SplitOpen, traceToLogsOptions?
     // the moment. Issue is that the trace itself isn't clearly mapped to dataFrame (right now it's just a json blob
     // inside a single field) so the dataLinks as config of that dataFrame abstraction breaks down a bit and we do
     // it manually here instead of leaving it for the data source to supply the config.
+    const query = getLinkForSplunk(span, traceToLogsOptions);
```

Comment:
We should move these calls inside the switch statement. At the moment, its running both functions when it should just run one of them based on the datasource type. Checking if `!expr` can also be inside the loki switch case